### PR TITLE
Feat: Email notification on secret change requests

### PR DESCRIPTION
**Pull request by DanielHougaard**
# Description 📣

This PR aims to add support for email notifications for secret change requests. Whenever a new secret change request is created, the reviewers will be notified via. email that their attention is needed.

The email will include a direct link to the change request.

Additionally: This PR also resolves an issue with creation secret change policies.

## Type ✨

- [x] Bug fix
- [x] New feature
- [ ] Breaking change
- [ ] Documentation

---

- [x] I have read the [contributing guide](https://infisical.com/docs/contributing/getting-started/overview), agreed and acknowledged the [code of conduct](https://infisical.com/docs/contributing/getting-started/code-of-conduct). 📝

### CODE REVIEW

File: backend/src/ee/services/secret-approval-request/secret-approval-request-fns.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+}: TSendApprovalEmails) => {
+  const cfg = getConfig();
+
+  const policy = await secretApprovalPolicyDAL.findById(secretApprovalRequest.policyId);
+
+  const project = await projectDAL.findProjectWithOrg(projectId);
+
+  // now we need to go through each of the reviewers and print out all the commits that they need to approve
+  for await (const reviewerUser of policy.userApprovers) {
+    if (reviewerUser) {
```
Comment (maidul98) [inline backend/src/ee/services/secret-approval-request/secret-approval-request-fns.ts:31]: Why would this be falsey? Won't this be a legit item

File: frontend/src/views/SecretApprovalPage/components/SecretApprovalRequest/SecretApprovalRequest.tsx [hunk 2]
```diff
@@ -66,10 +67,19 @@ export const SecretApprovalRequest = () => {
   const { user: userSession } = useUser();
   const { permission } = useProjectPermission();
   const { data: members } = useGetWorkspaceUsers(workspaceId);
-  const isSecretApprovalScreen = Boolean(selectedApproval);
+  const isSecretApprovalScreen = Boolean(selectedApprovalId);
+  const queryParams = new URLSearchParams(window.location.search);
```
Comment (maidul98) [inline frontend/src/views/SecretApprovalPage/components/SecretApprovalRequest/SecretApprovalRequest.tsx:71]: You can use the https://nextjs.org/docs/app/api-reference/functions/use-search-params instead

**Review by maidul98 — COMMENTED** [review]
Left some small comments

File: backend/src/ee/services/secret-approval-request/secret-approval-request-fns.ts (hunk 1 above)
Comment (DanielHougaard) [reply backend/src/ee/services/secret-approval-request/secret-approval-request-fns.ts:31]: This is true! Sorry this check is a result of an old implementation I was playing around with.

File: frontend/src/views/SecretApprovalPage/components/SecretApprovalRequest/SecretApprovalRequest.tsx (hunk 2 above)
Comment (DanielHougaard) [reply frontend/src/views/SecretApprovalPage/components/SecretApprovalRequest/SecretApprovalRequest.tsx:71]: @maidul98 I believe next/navigation was added in Next 13, we're still on version 12.3.

CleanShot 2024-08-26 at 20 52 36

Comment (akhilmhdh) [reply frontend/src/views/SecretApprovalPage/components/SecretApprovalRequest/SecretApprovalRequest.tsx:71]: Ohh actually useRouter has query component with it

Comment (DanielHougaard) [reply frontend/src/views/SecretApprovalPage/components/SecretApprovalRequest/SecretApprovalRequest.tsx:71]: @akhilmhdh I wonder why we use `URLSearchParams` everywhere else? I remember we discussed this a while back, I don't remember the exact reason for using `URLSearchParams` though 🤔

Comment (maidul98) [reply frontend/src/views/SecretApprovalPage/components/SecretApprovalRequest/SecretApprovalRequest.tsx:71]: nice catch, i see it here https://nextjs.org/docs/pages/api-reference/functions/use-router#router-object

**Review by akhilmhdh — APPROVED** [review]

**Pull request merged**